src/libraries/TickMath.sol

```diff
@@ -39,6 +39,21 @@ library TickMath {
         }
     }
 
+    /// @notice Check if sqrtPriceX96 still in `tick` after price movement
+    /// @dev Used to prevent costly `getTickAtSqrtPrice` calculation for small swaps
+    /// @param sqrtPriceX96 The fixed point Q64.96 value of the sqrt price after price movement
+    /// @param tick Tick ​​for which the check is carried out
+    /// @param nextNotCrossedTick The next active tick that is not crossed
+    /// @param zeroForOne The direction of price movement
+    function isSqrtPriceInTick(uint160 sqrtPriceX96, int24 tick, int24 nextNotCrossedTick, bool zeroForOne)
+        internal
+        pure
+        returns (bool)
+    {
+        if (zeroForOne) return nextNotCrossedTick == tick || sqrtPriceX96 >= getSqrtPriceAtTick(tick);
+        return nextNotCrossedTick == tick + 1 || sqrtPriceX96 < getSqrtPriceAtTick(tick + 1);
```

Comment:
```suggestion
        if (zeroForOne) return nextNotCrossedTick == tick ? true : sqrtPriceX96 >= getSqrtPriceAtTick(tick);
        unchecked {
            return nextNotCrossedTick == tick + 1 ? true : sqrtPriceX96 < getSqrtPriceAtTick(tick + 1);
        }
```
nit: make the short-circuit more explicit in case the compiler fails to optimize it.